### PR TITLE
cryptsetup-tests.py: add distribution specific test cases

### PR DESCRIPTION
**Pull request by nasastry**
Add distribution specific test cases.
With the support of yaml, can run distribution and upstream test cases.

Signed-off-by: Nageswara R Sastry <rnsastry@linux.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
RHEL:
```
JOB ID     : 0212787e40da08e5324029a0e7d97a2100101028
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-20T11.08-0212787/job.log
 (1/2) cryptsetup-tests.py:CryptSetup.test;run-run_type-upstream-583f:  PASS (2858.50 s)
 (2/2) cryptsetup-tests.py:CryptSetup.test;run-run_type-distro-70cf:  PASS (3054.48 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-20T11.08-0212787/results.html
JOB TIME   : 5921.07 s
```

**Comment by nasastry** [comment]
```
SLES:
JOB ID     : e48cd7bf691321ee1967c2fdfb452e60c2c7e745
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-07T12.28-e48cd7b/job.log
 (1/2) cryptsetup-tests.py:CryptSetup.test;run-run_type-upstream-a5fd:  PASS (2521.88 s)
 (2/2) cryptsetup-tests.py:CryptSetup.test;run-run_type-distro-bae5:  ERROR: cryptsetup source install failed. (4.82 s)
RESULTS    : PASS 1 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-07T12.28-e48cd7b/results.html
JOB TIME   : 2547.83 s


RHEL:
JOB ID     : 1336b5e02b05ffa2c7a040d395488ed0b8ea7524
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-07T02.08-1336b5e/job.log
 (1/2) cryptsetup-tests.py:CryptSetup.test;run-run_type-upstream-a5fd:  PASS (2656.67 s)
 
```

File: security/cryptsetup-tests.py [hunk 1]
```diff
@@ -34,33 +34,53 @@ def setUp(self):
...
         for package in deps:
             if not smm.check_installed(package) and not smm.install(package):
                 self.cancel('%s is needed for the test to be run' % package)
-        url = "https://gitlab.com/cryptsetup/cryptsetup/"
-        git.get_repo(url, destination_dir=self.workdir)
-        os.chdir(self.workdir)
-        process.run("./autogen.sh")
-        process.run("./configure")
-        build.make(self.workdir)
+        run_type = self.params.get('type', default='upstream')
```
Comment (Naresh-ibm) [inline security/cryptsetup-tests.py:50]: @nasastry make distro as default parameter

File: security/cryptsetup-tests.py [hunk 2]
```diff
@@ -34,33 +34,53 @@ def setUp(self):
...
+        if run_type == "upstream":
+            default_url = "https://gitlab.com/cryptsetup/cryptsetup/"
```
Comment (Naresh-ibm) [inline security/cryptsetup-tests.py:52]: remove the default_url as we are hard_coding it in yaml

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

File: security/cryptsetup-tests.py (hunk 2 above)
Comment (abdhaleegit) [reply security/cryptsetup-tests.py:52]: @Naresh-ibm the idea here is to make sure the test runs with out yaml.. so we are good here

File: security/cryptsetup-tests.py (hunk 1 above)
Comment (abdhaleegit) [reply security/cryptsetup-tests.py:50]: as far as community code is concerned always good to have upstream default.. we can use yaml to use distro first in CR

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Review by Naresh-ibm — APPROVED** [review]
agreed to have default as upstream and hard_coded default url.
So LGTM

**Pull request merged**